tests/base/test_tensor_innerproduct.py

```diff
@@ -162,6 +162,166 @@ def test_order6_faces_invert_model(self):
         self.orderTest()
 
 
+class TestInnerProductsFaceProperties3D(discretize.tests.OrderTest):
+    """Integrate a function over a surface within a unit cube domain
+    using edgeInnerProducts and faceInnerProducts."""
+
+    meshTypes = ["uniformTensorMesh"]
+    meshDimension = 3
+    meshSizes = [16, 32]
+
+    def getError(self):
+        call = lambda fun, xyz: fun(xyz[:, 0], xyz[:, 1], xyz[:, 2])  # NOQA E731
+
+        ex = lambda x, y, z: x**2 + y * z  # NOQA E731
+        ey = lambda x, y, z: (z**2) * x + y * z  # NOQA E731
+        ez = lambda x, y, z: y**2 + x * z  # NOQA E731
+
+        tau_x = lambda x, y, z: y * z + 1  # NOQA E731  # x-face properties
+        tau_y = lambda x, y, z: x * z + 2  # NOQA E731  # y-face properties
+        tau_z = lambda x, y, z: 3 + x * y  # NOQA E731  # z-face properties
+
+        tau = 3 * [None]
+        for ii, comp in enumerate(["x", "y", "z"]):
+            k = np.isclose(
+                eval("self.M.faces_{}".format(comp))[:, ii], 0.5
+            )  # x, y or z location for each plane
+            tau_ii = 1e-8 * eval(
+                "np.ones(self.M.nF{})".format(comp)
+            )  # effectively zeros but stable
+            tau_ii[k] = eval("call(tau_{}, self.M.faces_{}[k, :])".format(comp, comp))
```

Comment:
I think you could effectively test this with only a single function for `tau` instead of defining three different ones for each cell face, and then having to use this very odd line to evaluate everything. You should try that and see if it also works.